src/averageAges.js

```diff
@@ -52,7 +77,30 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const result = [];
+  let averageAge;
+
+  if (!onlyWithSon) {
+    people.forEach(mother =>
+      people.map(child => {
+        if (mother.name === child.mother) {
+          result.push(child.born - mother.born);
+        }
+      })
+    );
+    averageAge = result.reduce((accum, elem) => accum + elem, 0);
+  } else {
+    people.forEach(mother =>
+      people.forEach(son => {
+        if (mother.name === son.mother && son.sex === 'm') {
+          result.push(son.born - mother.born);
+        }
+      })
+    );
+    averageAge = result.reduce((accum, elem) => accum + elem, 0);
+  }
+
+  return averageAge / result.length;
```

Comment:
I think averageAge is not the best naming for this var, cause it's the sum of all ages



return